### PR TITLE
TESTING: Homepage/Map View

### PR DESCRIPTION
**Pull request by sam-rice**
## Summary

This branch adds a Cypress spec for the homepage/map view and supporting fixture.

### Details

Testing covers components:
- `MapView` (comprehensive)
- `Header` (partial)
- `PopupContent` (comprehensive)

### Notes to Reviewer

- The additions to cypress.config.ts, commands.ts, and index.d.ts were intended for adding a custom Cypress command. After researching for a while and trying different suggestions, I couldn't find a way to avoid a type error, so the command isn't being used for now. [The issue](https://github.com/cypress-io/cypress-documentation/issues/2565) seems to be somewhat common and ongoing.
- `cy.wait()`s were added in all `beforeEach`s to account for the lagginess of Leaflet. Tests were randomly failing before adding the delay between them.
- Any tips on how to clean up the lengthier syntax or anything else would be much appreciated!

### CODE REVIEW

File: cypress/e2e/MapView_spec.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      {
+        method: "GET",
+        url: "http://localhost:3001/v1/trees",
+      },
+      {
+        fixture: "trees.json",
+      }
+    )
+    cy.visit("http://localhost:3000")
+    cy.wait(1000)
```
Comment (Adelle-Pitsas) [inline cypress/e2e/MapView_spec.cy.tsx:15]: So I've been looking into why leaflet is so laggy and a possible solution to the wait situation. Honestly, it looks like if it works, then waiting seems like the way to go. Found this article that discusses this issue with some solutions: https://filiphric.com/waiting-in-cypress-and-how-to-avoid-it

File: cypress/e2e/MapView_spec.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  })
+
+  it("should display the site title", () => {
+    cy.get('[data-cy="site-title"]').should("have.text", "TREESOFSEATTLE")
+  })
+
+  it("should have a button for navigating to the \"new tree\" form", () => {
+    cy.get('[data-cy="new-tree-button"]').click()
+    cy.url().should("eq", "http://localhost:3000/new-tree")
```
Comment (Adelle-Pitsas) [inline cypress/e2e/MapView_spec.cy.tsx:24]: Nice job testing the path to where a button should take you!

File: cypress/e2e/MapView_spec.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        fixture: "trees.json",
+      }
+    )
+    cy.visit("http://localhost:3000")
+    cy.wait(1000)
+    cy.get('[data-cy="marker-filter"]').type("doug")
+  })
+
+  it("should have a text input for filtering tree map markers by species name", () => {
+    cy.get('[data-cy="marker-filter"]').should("have.value", "doug")
```
Comment (Adelle-Pitsas) [inline cypress/e2e/MapView_spec.cy.tsx:101]: Nice--testing value of typed input

**Comment by Adelle-Pitsas** [comment]
This all looks awesome and I learned a lot from reviewing your code! Everything looks super clean and simple. Super excited to see the final product!

**Review by Adelle-Pitsas — APPROVED** [review]

File: cypress/e2e/MapView_spec.cy.tsx (hunk 1 above)
Comment (sam-rice) [reply cypress/e2e/MapView_spec.cy.tsx:15]: This is a great resource! Thanks for looking that up. I might play around with some of the suggestions in there.

**Pull request merged**